app/views/documents/_header.html.erb

```diff
@@ -24,3 +24,8 @@
                                  topics: topics,
                                  primary_mainstream_category: primary_mainstream_category,
                                  specialist_tag_finder: specialist_tag_finder %>
+
+<% if document.historic? %>
+<%= render('documents/history_notice', type: document.format_name,
+                                       government_name: document.government.try(:name)) %>
```

Comment:
I hear what you're saying, but I think this has less to do with method implementation and more to do with how we are modelling the domain. Right now, the way we model things is that an historic document will always have a government (down to the fact that we cannot determine the government of a yet-to-be-published document without a publication date so we have to assume it is not historic). In theory, some point in the future it may be possible to have an `historic?` document that is not associated with a government. But writing code to handle the maybe-one-day future also feels like bad practice to me (YAGNI and all that). The cost of it confusing people right now (making it seem like an historic document might be historic without having a government) is higher than the potential cost of (maybe) having to change the partial to handle `nil` government if the modelling changes in the future.